desk/app/reel.hoon

```diff
@@ -79,7 +80,20 @@
   =/  old  !<(versioned-state old-state)
   ?-  -.old
       %4
-    `this(state old)
+    =.  state  old
+    =^  new-md  stable-id
+      %+  roll
+        ~(tap by our-metadata)
+      |=  [[=token:reel =metadata:reel] [md=_our-metadata id=_stable-id]]
+      ?^  (slaw %uv token)  [md id]
+      ?^  (rush token flag)
+        :-  md
+        ?:  (~(has by id) token)  id
+        (~(put by id) token token)
+      =/  new  (rap 3 (scot %p our.bowl) '/' token ~)
+      :-  (~(put by md) new metadata)
+      (~(put by id) new new)
```

Comment:
This is fine to run on every load, because none of the transformations result in follow-up:
- New-style tokens (`@uv` format) are untouched,
- "New old-style" tokens (`~ship/term` format) only guarantee they have an entry in `stable-id`,
- "Old old-style" tokens get upgraded into "new old-style" tokens and put an entry in `stable-id`.

Correct?

Reply:
(I'd still feel more comfortable with migration moving into state version `%5`, fwiw, just because it doesn't even risk repeated transforms.)

Reply:
yep!